src/views/AutogenView.vue

```diff
@@ -381,10 +382,19 @@ export default {
   watch: {
     '$route' (to, from) {
       if (to.fullPath !== from.fullPath && !to.fullPath.includes('action/')) {
-        this.searchQuery = ''
-        this.page = 1
+        if ('q' in to.query) {
```

Comment:
should pick up `query.keyword` instead ?

Reply:
I think not all views/APIs pass keyword as the query parameter (from search); for example global setting we use name

Reply:
Agreed but it creates two parameters keywork / name and q, with q having the higher priority, so changing only q and not the other can lead to inconsistencies. It by default picks up keywork / name when q is not present anyway